src/Core/Silk.NET.BuildTools/Converters/Readers/OpenCLReader.cs

```diff
@@ -835,42 +755,181 @@ public IEnumerable<Enum> ReadEnums(object obj, BindTask task)
-                .DistinctBy(x => x.Attribute("name")?.Value)
                 .Where
                 (
-                    x => x.Parent?.Attribute("name")?.Value != "Constants" &&
-                         x.Parent?.Attribute("name")?.Value != "MiscNumbers"
+                    x => !x.Parent.Attribute("name").Value.StartsWith("Constants") &&
+                         !x.Parent.Attribute("name").Value.StartsWith("enums") &&
+                         x.Parent.Attribute("name").Value != "MiscNumbers" &&
+                         x.Parent.Attribute("name").Value != "cl_device_info"
                 )
                 .ToDictionary
                 (
-                    x => x.Attribute("name")?.Value,
+                    x => Rename(x.Attribute("name").Value, task),
+                    x =>
+                    {
+                        var type = Rename(x.Parent.Attribute("name").Value, task);
+                        string newName = "";
+                        var typeStr = type.Renamed;
+                        if (typeStr.StartsWith("ErrorCodes"))
+                        {
+                            return new RenamedEntry(type.Original, "ErrorCodes");
+                        }
+                        if (typeStr.EndsWith(".flags"))
+                        {
+                            newName = typeStr.Substring(0, typeStr.Length - ".flags".Length);
+                            flagEnums.Add(newName);
+                            return new RenamedEntry(type.Original, newName);
+                        }
+                        newName = typeStr.Replace('.', '_');
+                        if (flagEnums.Contains(typeStr))
+                        {
+                            flagEnums.Add(newName);
+                        }
+                        return new RenamedEntry(type.Original, newName);
+                    }
+                );
+
+            apis.Elements("require")
+                .Elements("enum").ForEach(x =>
+                {
+                    var name = Rename(x.Attribute("name").Value, task);
+
+                    if (constants.Contains(name)) return;
+
+                    var type = Rename(x.Parent.Attribute("comment")?.Value, task);
+
+                    var typeStr = type?.Renamed;
+
+                    if (typeStr == null || typeStr.Contains(' '))
+                    {
+                        if (enumsPass1.TryGetValue(name, out var pass1type))
+                        {
+                            type = pass1type;
+                        }
+                        else
+                        {
+                            typeStr = name.Renamed switch
+                            {
+                                "CL_CONTEXT_MEMORY_INITIALIZE_KHR" => "cl_context_properties",
+                                "CL_DEVICE_PARTITION_BY_NAMES_INTEL" => "cl_device_partition_property",
+                                _ => typeStr
+                            };
```

Comment:
Can these not be hardcoded

Reply:
Well, yeah, I've been thinking about better ways but...
Probably adding it to JSON spec same as renaming pass could be an option.